client/history/client.go

```diff
@@ -1053,21 +1054,34 @@ func (c *clientImpl) GetReplicationStatus(
 	request *historyservice.GetReplicationStatusRequest,
 	opts ...grpc.CallOption,
 ) (*historyservice.GetReplicationStatusResponse, error) {
-	// TODO: fan out this request to all history services
-	client, err := c.getClientForShardID(1)
-	var response *historyservice.GetReplicationStatusResponse
-	op := func(ctx context.Context, client historyservice.HistoryServiceClient) error {
-		var err error
-		ctx, cancel := c.createContext(ctx)
-		defer cancel()
-		response, err = client.GetReplicationStatus(ctx, request, opts...)
-		return err
+	clients, err := c.clients.GetAllClients()
+	if err != nil {
+		return nil, err
 	}
-	err = c.executeWithRedirect(ctx, client, op)
+
+	eg, ctx := errgroup.WithContext(ctx)
+	lock := sync.Mutex{}
+	result := &historyservice.GetReplicationStatusResponse{}
+
+	for _, client := range clients {
+		historyClient := client.(historyservice.HistoryServiceClient)
+		eg.Go(func() error {
+			resp, err := historyClient.GetReplicationStatus(ctx, request, opts...)
+			if err != nil {
+				return err
+			}
+			lock.Lock()
+			defer lock.Unlock()
+			result.Shards = append(result.Shards, resp.Shards...)
+			return nil
+		})
+	}
+	err = eg.Wait()
 	if err != nil {
 		return nil, err
```

Comment:
Should it return what it has in the response?

Reply:
I think it does not matter.